ui-asset-management/api/svc-asset-details/extension/clients/secrets_client.go

```diff
@@ -20,53 +20,86 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
 	"svc-asset-details-layer/models"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
 )
 
 type SecretsClient struct {
-	secretsClient *secretsmanager.SecretsManager
+	secretsClient *secretsmanager.Client
 }
 
 var (
 	versionStage string = "AWSCURRENT"
 )
 
-func NewSecretsClient(region string) *SecretsClient {
-	svc := secretsmanager.New(
-		session.New(),
-		aws.NewConfig().WithRegion(region),
-	)
+// NewSecretsClient initializes the SecretsManager client
+func NewSecretsClient(assumeRoleArn, region string) (*SecretsClient, error) {
+	// Load the default configuration with region
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(region))
+	if err != nil {
+		fmt.Printf("error loading AWS config: %v", err)
+		return nil, err
+	}
+
+	// Create an STS client
+	stsClient := sts.NewFromConfig(cfg)
+
+	// Assume the role using STS
+	creds := stscreds.NewAssumeRoleProvider(stsClient, assumeRoleArn, func(o *stscreds.AssumeRoleOptions) {
+		o.RoleSessionName = "SecretsSession"
+	})
+
+	// Create a new configuration with the assumed role credentials
+	assumedCfg := aws.Config{
+		Credentials: aws.NewCredentialsCache(creds),
+		Region:      region,
+	}
+
+	// Initialize the Secrets Manager client
+	svc := secretsmanager.NewFromConfig(assumedCfg)
 
 	fmt.Println("Initialized Secrets Client")
 	return &SecretsClient{
 		secretsClient: svc,
-	}
+	}, nil
 }
 
-func (r *SecretsClient) GetRdsSecret(ctx context.Context, secretName string) (*models.RdsSecret, error) {
+// GetRdsSecret retrieves the RDS secret from AWS Secrets Manager
+func (r *SecretsClient) GetRdsSecret(ctx context.Context, secretIdPrefix, tenantId string) (*models.RdsSecret, error) {
+	// Create the secretId using the prefix and tenantId
+	secretId := fmt.Sprintf("{%s}{%s}", secretIdPrefix, tenantId)
+	fmt.Printf("Getting Rds Secrets from {%s}\n", secretId)
+
+	// Prepare the input for retrieving the secret
 	input := &secretsmanager.GetSecretValueInput{
-		SecretId:     aws.String(secretName),
+		SecretId:     aws.String("paladincloud/secret/98c28482-9bae-46bd-bd4e-58fa132e72c0"),
```

Comment:
Did you mean to replace the hard-coded string with `secretId` that was created a few lines above?